crates/world/src/conversions.rs

```diff
@@ -218,3 +218,179 @@ mod llm {
         }
     }
 }
+
+mod observe {
+    use super::*;
+    use opentelemetry::StringValue;
+    use std::time::{Duration, SystemTime, UNIX_EPOCH};
+    use wasi::observe::tracer;
+
+    impl From<tracer::Value> for opentelemetry::Value {
+        fn from(value: tracer::Value) -> Self {
+            match value {
+                tracer::Value::String(v) => v.into(),
+                tracer::Value::Bool(v) => v.into(),
+                tracer::Value::Float64(v) => v.into(),
+                tracer::Value::S64(v) => v.into(),
+                tracer::Value::StringArray(v) => opentelemetry::Value::Array(
+                    v.into_iter()
+                        .map(StringValue::from)
+                        .collect::<Vec<_>>()
+                        .into(),
+                ),
+                tracer::Value::BoolArray(v) => opentelemetry::Value::Array(v.into()),
+                tracer::Value::Float64Array(v) => opentelemetry::Value::Array(v.into()),
+                tracer::Value::S64Array(v) => opentelemetry::Value::Array(v.into()),
+            }
+        }
+    }
+
+    impl From<tracer::KeyValue> for opentelemetry::KeyValue {
+        fn from(kv: tracer::KeyValue) -> Self {
+            opentelemetry::KeyValue::new(kv.key, kv.value)
+        }
+    }
+
+    impl From<tracer::TraceFlags> for opentelemetry::trace::TraceFlags {
+        fn from(flags: tracer::TraceFlags) -> Self {
+            Self::new(flags.as_array()[0] as u8)
+        }
+    }
+
+    impl From<opentelemetry::trace::TraceFlags> for tracer::TraceFlags {
+        fn from(flags: opentelemetry::trace::TraceFlags) -> Self {
+            if flags.is_sampled() {
+                tracer::TraceFlags::SAMPLED
+            } else {
+                tracer::TraceFlags::empty()
+            }
+        }
+    }
+
+    impl From<tracer::SpanContext> for opentelemetry::trace::SpanContext {
+        fn from(sc: tracer::SpanContext) -> Self {
+            // TODO(Reviewer): Should this be try_from instead an propagate this error out of the WIT?
```

Comment:
Some of these conversions end up being fallible b/c we're parsing things that must conform to a format. Right now I handle this by just quelling the error and silently omitting the invalid data. This doesn't feel like a great user experience (even if it is documented in the WIT interface).

An alternative would be to make this `try_from` and then expose the error out through the WIT interface. This would make change:
- `start: func(name: string, options: option<start-options>) -> span;` to `start: func(name: string, options: option<start-options>) -> result<span>;`
- `add-link: func(link: link);` to `add-link: func(link: link) -> result<()>;`

I don't love the idea of exposing errors out of these functions just for the incorrect parsing of a span context. I'm not sure what the right choice is here.